packages/offix-datastore/src/utils/NetworkStatus.ts

```diff
@@ -0,0 +1,38 @@
+import Observable from "zen-observable";
+import { SubscriptionClient } from "subscriptions-transport-ws";
+
+export interface NetworkEvent {
+  /**
+   * Network event type
+   */
+  event: string;
+  /**
+   * Connection status
+   */
+  status: boolean;
+};
+
+export class NetworkStatus {
```

Comment:
Would we also drop some interface in future - maybe separate issue to allow people override that if needed? Def not target of the issue. Would that be possible to be introduced at some point?

Reply:
Yeah I will create an issue for this

Reply:
Actually, I just created and exposed an interface, this is a quick win. That and the enum.

The unsubscribing from the listener is a larger issue to tackle and we can do the tests along with this.